### PR TITLE
add guidelines for reporting security issues (close #322)

### LINKED ISSUE #322: add guidelines to report security issues
README has to be updated with a section that lays out guidelines to report security issues, preferably an email address.

### PR DESCRIPTION
**Pull request by vipulgupta2048**
- Reviewed information from [1](https://github.com/shark-h/graphql-engine/tree/sec_info#contributing), [2](https://github.com/shark-h/graphql-engine/blob/6f3130756ea84c9f031a1dd948f1270c4d2af96f/.github/SECURITY.md) and even [3](https://github.com/shark-h/graphql-engine/blob/6f3130756ea84c9f031a1dd948f1270c4d2af96f/.github/SECURITY.md)
- Proposed guidelines for reporting securtiy issues.
- Polished it a bit. 
- Linked it back to `README.md`
- Solves #322 
 
**Questions**
- Should we have any prizes for people who report security issues? Swag or anything like bug bounty.
- Would you like to have any format in which people should report vulnerablities?
- Could this count under #hacktoberfest
- I have edited the email tag - hence one could also add a subject line if needed? Want to?
- Unsure about keeping this in `.github` folder

Signed-off-by: Vipul Gupta (@vipulgupta2048) <vipulgupta2048@gmail.com>

### CODE REVIEW

File: .github/SECURITY.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+- You need help tuning Hasura GraphQL Engine components for security.
+- You need help applying security related updates.
+- Your issue is not security related.
+
+## Security Vulnerability Response
+
+Each report is acknowledged and analyzed by Product Security Team members within 3 working days. 
+
+As the security issue moves from triage, to identified fix, to release planning we will keep the reporter updated.
```
Comment (dsandip) [inline .github/SECURITY.md]: The reporter will be kept updated at every stage of the issue's analysis and resolution (triage -> fix -> release).

File: .github/SECURITY.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+## Public Disclosure Timing
+
+A public disclosure date is negotiated by the Hasura GraphQL product security team and the bug submitter. We prefer to fully disclose the bug as soon as possible once a user mitigation is available. It is reasonable to delay disclosure when the bug or the fix is not yet fully understood, the solution is not well-tested, or for vendor coordination. The timeframe for disclosure is from immediate (especially if it's already publicly known) to a few weeks. As a basic default, we expect report date to disclosure date to be on the order of 7 days. The Hasura GraphQL product security team holds the final say when setting a disclosure date.
```
Comment (dsandip) [inline .github/SECURITY.md]: 1) Change`As a basic default, we expect report date to disclosure date to be on the order of 7 days.` to `We expect the time-frame between a report to a public disclosure to typically be in the order of 7 days.`
2) Change `The Hasura GraphQL product security team holds the final say when setting a disclosure date.` -> `The Hasura GraphQL Engine maintainers and the security team will take the final call on setting a disclosure date.`

File: .github/SECURITY.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+
+Credits: Some sections are adapted from [https://github.com/kubernetes/website/blob/master/content/en/docs/reference/issues-security/security.md](https://github.com/kubernetes/website/blob/master/content/en/docs/reference/issues-security/security.md)
```
Comment (dsandip) [inline .github/SECURITY.md]: Credits: Some sections have been inspired and adapted from [https://github.com/kubernetes/website/blob/master/content/en/docs/reference/issues-security/security.md](https://github.com/kubernetes/website/blob/master/content/en/docs/reference/issues-security/security.md).

File: .github/SECURITY.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
+## Reporting Vulnerabilities
+
+We’re extremely grateful for security researchers and users that report vulnerabilities to the Hasura GraphQL Community. All reports are thoroughly investigated by a set of community volunteers.
+
+To make a report, please email us at [build@hasura.io](mailto:build@hasura.io) with the complete security details attaching all necessary information.
```
[the remark this answers is not in the document]
Comment (vipulgupta2048) [reply .github/SECURITY.md]: Implemented.

File: .github/SECURITY.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+### When Should I Report a Vulnerability?
+
+- You think you discovered a potential security vulnerability in the Hasura GraphQL Engine or related components.
```
[the remark this answers is not in the document]
Comment (vipulgupta2048) [reply .github/SECURITY.md]: Good catch. Implemented.

File: .github/SECURITY.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+- You are unsure how a vulnerability affects the Hasura GraphQL Engine.
+- You think you discovered a vulnerability in another project that Hasura GraphQL Engine depends on (e.g. heroku, docker, etc)
+- Any other risk that would harm the Hasura GraphQL Engine.
```
[the remark this answers is not in the document]
Comment (vipulgupta2048) [reply .github/SECURITY.md]: Sounds better, implemented.

**Comment by vipulgupta2048** [comment]
@dsandip Made all changes as requested. Document is now at root, with the same. The README file has also been updated. 

> Yes, can you add a section called "Credits" and call out the fact that while we'll be coming up with a bug bounty at a later stage, we'd like to publically acknowledge reporters and send them some Hasura swag.

Adding Credits to a security guidelines document seems odd. This section would look better on the website or in another document. What say?

@shahidhk Can you add the tag #hacktoberfest :heart_eyes:

**Comment by vipulgupta2048** [comment]
@dsandip please merge this pull request. Or if you have merged commits into the master then comment here and close the PR

**Comment by dsandip** [comment]
@vipulgupta2048 Sorry about the delay on the review. You were right the first time around - the credits/what-do-I-get section looks very off!. The rest of it looks good, can you please remove that commit and resubmit. Thanks!

**Pull request merged**